### PR TITLE
Add tests for abstract graph basic methods

### PR DESCRIPTION
**Pull request by shakareem**
Add parameterized tests for all basic methods of abstract graph, they are tested on all its inheritors.

### CODE REVIEW

**Review by qrutyy — APPROVED** [review]

File: app/src/test/kotlin/model/abstractGraph/GraphTest.kt [hunk 1]
```diff
@@ -27,16 +26,271 @@ fun provideAllGraphTypes(): Stream<Arguments> {
...
+
+                    val actualVertices = graph.getVertices()
+                    val expectedVertices = listOf(
+                        Vertex(0, 0),
+                        Vertex(1, 1),
+                        Vertex(2, 4),
+                        Vertex(3, 3),
+                    )
+
+                    assertEquals(expectedVertices, actualVertices)
```
Comment (spisladqo) [inline app/src/test/kotlin/model/abstractGraph/GraphTest.kt]: It doesn't work as expected, please remake it everywhere it appears .

**Review by spisladqo — CHANGES_REQUESTED** [review]
Sadly, every parametrized test falls.

And some of the assertions will always be false, because Vertex and Edge are classes, but not data classes, so `equals()` compares hashes of these objects. I don't think we need to make them data classes, because tests can be remade to work as expected.

**Comment by shakareem** [comment]
> Sadly, every parametrized test falls.
> 
> And some of the assertions will always be false, because Vertex and Edge are classes, but not data classes, so `equals()` compares hashes of these objects. I don't think we need to make them data classes, because tests can be remade to work as expected.

Very helpful remarks, thank you!! Will try to fix it ASAP!!

**Comment by shakareem** [comment]
Parameterized tests work as it was intended. Found a bug in removeVertexFromEverywhere method, we are trying to change set while iterating over it. Also some tests don't work as intended. TODO: fix this bug and tests.

**Comment by shakareem** [comment]
FIxed found bug in another PR, found new bug))). Everything in tests work as intended. PR is ready to be merged.
TODO: fix found bug.

**Review by qrutyy — APPROVED** [review]
Nice! 
As i understood, the found bug is already fixed in another PR, but in TODO you mentioned to fix another one?
Pls comment, what bug you need to fix)

**Comment by shakareem** [comment]
> Nice! 
> As i understood, the found bug is already fixed in another PR, but in TODO you mentioned to fix another one?
> Pls comment, what bug you need to fix)

Yes, there is another one, will fix it in the same #26 PR.

The bug is: when fixing ID fragmentation, we don't add copied edges to the edges property.

**Review by spisladqo — APPROVED** [review]
Good job!

**Pull request merged**